index.js

```diff
@@ -71,7 +77,35 @@ function registerListener(session) {
 			item.setSavePath(filePath);
 		}
 
-		item.on('updated', () => {
+		item.on('updated', (e, state) => {
+      if (handlers.retryCount === 3) {
```

Comment:
Move this limit to config, also use >= to capture the possible case that it may be incremented before you can check it.